src/main/networking/vtunnel.ts

```diff
@@ -31,22 +37,48 @@ export class vtunnel {
     },
   });
 
+  constructor() {
+    if (vtunnel._instance) {
+      throw new Error('Error: Use vtunnel.getInstance() instead of calling new.');
+    }
+    vtunnel._instance = this;
+  }
+
+  public static getInstance(): vtunnel {
+    return vtunnel._instance;
+  }
+
+  addTunnel(config: VtunnelConfig) {
+    this._vtunnelConfig.push(config);
+  }
+
   async generateVtunnelConfig() {
```

Comment:
Presumably this is never going to generate non-vtunnel configs, so calling it just `generateConfig` is probably fine?

Any reason why we don't just always call this in `start` (and make this method `protected`)?